File: sqry-lang-rust/tests/fixtures/localvars/scoping.rs
// Scoping and loop variable tests

fn shadowed_var() {
    let x = 1;
    {
        let x = 2;
        println!("{}", x);
    }
    println!("{}", x);
}

fn for_loop_var() {
    let items = vec![1, 2, 3];
    for item in items {
        println!("{}", item);
    }
}

fn while_loop_var() {
    let mut count = 0;
    while count < 10 {
        println!("{}", count);
        count += 1;
    }
}

fn multiple_refs() {
    let x = 1;
    let y = x + x;
    let z = x + y;
    println!("{}", z);
}

File: sqry-lang-typescript/tests/fixtures/localvars/scoping.ts
// Scoping, shadowing, and loop variable tests

function shadowedVar(): void {
    let x = 10;
    console.log(x);
    {
        let x = 20;
        console.log(x);
    }
    console.log(x);
}

function forLoopVar(): void {
    for (let i = 0; i < 10; i++) {
        console.log(i);
    }
}

function forOfVar(): void {
    const items = [1, 2, 3];
    for (const item of items) {
        console.log(item);
    }
}

function forInVar(): void {
    const obj = { a: 1, b: 2 };
    for (const key in obj) {
        console.log(key);
    }
}

function multipleRefs(): void {
    let x = 1;
    let y = x + x;
    let z = x + y;
    console.log(z);
}
